### ISSUE TITLE
`:doc` should work for reserved names too

### ISSUE THREAD

**Issue by gallais** [idris-lang/Idris2#651]
At the moment `:doc` takes a `name` which explicitly puts reserved keywords
outside of its scope. This means that if we want to know more about `Int` and
type `:doc Int` we are left with a puzzling parse error when everything works
fine for e.g. `:doc Nat`.

We should fix this by:

* not using `name` as the parser for the argument to `:doc`
* hard-wiring doc for the reserved names that correspond to built-in things

**Comment by michaelmesser** [comment]
This looks simple to implement, but I have a few questions:
1. Currently the descriptions printed by `:doc` are a tab plus a space indented. Is that extra space intentional or should the leading white space be trimmed?
2. I plan to make a `Constant -> Maybe String` to get the description of a `Constant`. Should that go in `Core.TT` near the definition of `Constant` or in `Idris.DocString`? Should I call it `constantDocDescription`?
3. Are there already written descriptions that I can use?
4. Should I implement `:doc Type` too?

**Comment by gallais** [comment]
1. Don't know. It does not matter as long as the indentation is consistent IMO
2. Probably in `DocString`. I don't have opinions on the name.
3. Don't know.
4. Yes.

**Comment by michaelmesser** [comment]
What should `:doc 1`, `:doc ()`, and `:doc []` print? Trying to figure out if I should pattern match on `PTerm` or `RawImp`.

**Comment by michaelmesser** [comment]
What should `:doc ()` and `:doc 1` print? `fromInteger` and `IAlternative` make things confusing. I think it makes sense to handle things at the `TTImp` level rather than `PTerm` so that things like `:doc []` just work.

**Comment by gallais** [comment]
If things are overloaded, I think it's legitimate that `:doc` should say so and explain
how. For "builtin" overloading mechanisms like `fromInteger` and `fromString` it's worth
explaining that this is the way numeric and string literals are handled.

For `:t` it's a bit more complicated: typechecking an arbitrary expression requires
elaboration & disambiguation. I suppose we could manually add special cases to handle
isolated identifiers that happen to be ambiguous but realistically we can't do much
more for arbitrary expressions.

**Comment by gallais** [comment]
This was fixed in #1178. These days we even get the interfaces satisfied by these primitive types so instead
of a parse error, `:doc Int` will give you:

```
Main> :doc Int
Int : Type
  Primitive type of bounded signed integers (backend dependent size)
Hints:
  Abs Int
  Cast Int String
  Cast Int Integer
  Cast Integer Int
  Cast Char Int
  Cast Double Int
  Cast String Int
  Cast Nat Int
  Cast Bits8 Int
  Cast Bits16 Int
  Cast Bits32 Int
  Cast Bits64 Int
  Cast Int8 Int
  Cast Int16 Int
  Cast Int32 Int
  Cast Int64 Int
  Cast Int Char
  Cast Int Double
  Cast Int Bits8
  Cast Int Bits16
  Cast Int Bits32
  Cast Int Bits64
  Cast Int Int8
  Cast Int Int16
  Cast Int Int32
  Cast Int Int64
  Cast Int Nat
  Eq Int
  Integral Int
  Neg Int
  Num Int
  Ord Int
  Show Int
```

Fantastic!
